keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/Differentiator.java

```diff
@@ -44,46 +43,45 @@ public static PartialDerivatives reverseModeAutoDiff(Vertex<?> ofVertex, Partial
                 continue;
             }
 
-            if (visiting instanceof Differentiable) {
-                Differentiable visitingDifferentiable = ((Differentiable) visiting);
-                Map<Vertex, PartialDerivatives> partialDerivatives = visitingDifferentiable.reverseModeAutoDifferentiation(dwrtOf.get(visiting));
-                collectPartials(partialDerivatives, dwrtOf, visiting.getShape());
-            }
-
             if (!visiting.isProbabilistic()) {
-                for (Vertex parent : visiting.getParents()) {
-                    if (!alreadyQueued.contains(parent) && parent instanceof Differentiable) {
-                        priorityQueue.offer(parent);
-                        alreadyQueued.add(parent);
+
+                if (visiting instanceof Differentiable) {
```

Comment:
Would be worth adding a helper `.isDifferentiable` on to Vertex just like we have for `.isProbabilistic()`

Reply:
Once upon a time instanceof was considered slow. These days it's not a problem and I don't think worth refactoring due to it not being as simple as Probabilistic/NonProbabilistic.